components/dispatcher/internal/client/client.go

```diff
@@ -1,128 +1,102 @@
 
-func sendKafkaConsumerMessageToSubscriber(runner goresilience.Runner, hc httpClient, message []byte, logger *zap.Logger) error {
 	err := runner.Run(context.TODO(), func(_ context.Context) error {
-
-		// Send The Kafka ConsumerMessage To Subscriber URI Via HTTP POST Request
-		response, err := client.Post(hc.uri, "application/json", bytes.NewBuffer(message))
+		rctx, _, err := client.cloudEventClient.Send(context.Background(), event)
 		if err != nil {
-			logger.Warn("Failed to send message to subscriber service, retrying", zap.Error(err))
-			return err
-		}
-
-		// Log The Response & Defer Closing Response Body
-		if response != nil {
-
-			defer func() {
-
-				closeResponseBody(response, logger)
-
-			}()
-
-			// required to read the body (even though we don't need it) to avoid memory leaks when reusing http client:
-			_, _ = io.Copy(ioutil.Discard, response.Body)
-
-			err = logResponse(logger, response)
-			if err != nil {
-				return err
-			}
-
-		} else {
-			logger.Warn("Received nil response when sending message to subscriber service, retrying")
-			return errors.New("Received nil response when sending message to subscriber service")
+			transportContext := cloudevents.HTTPTransportContextFrom(rctx)
+			return logResponse(logger, transportContext.StatusCode)
 		}
-
 		return nil
 	})
-	return err
-}
 
-func closeResponseBody(response *http.Response, logger *zap.Logger) {
-	if response.Body != nil {
-		err := response.Body.Close()
-		if err != nil {
-			logger.Warn("Failed to close HTTP response body", zap.Error(err))
-		}
+	// Retries failed
+	if err != nil {
+		logger.Error("Failed to send after configured number of retries", zap.Error(err))
+		return err
 	}
+	return nil
 }
 
-func logResponse(logger *zap.Logger, response *http.Response) error {
-	if response.StatusCode >= 500 || response.StatusCode == 404 || response.StatusCode == 429 {
-		logger.Warn("Failed to send message to subscriber service, retrying", zap.String(responseStatus, response.Status))
-		return errors.New("Server returned a bad response code: " + strconv.Itoa(response.StatusCode))
-	} else if response.StatusCode > 299 {
-		logger.Warn("Failed to send message to subscriber service, not retrying", zap.String(responseStatus, response.Status))
+func logResponse(logger *zap.Logger, statusCode int) error {
+	if statusCode >= 500 || statusCode == 404 || statusCode == 429 {
+		logger.Warn("Failed to send message to subscriber service, retrying", zap.Int("statusCode", statusCode))
+		return errors.New("Server returned a bad response code: " + strconv.Itoa(statusCode))
+	} else if statusCode > 299 {
+		logger.Warn("Failed to send message to subscriber service, not retrying", zap.Int("statusCode", statusCode))
 	} else {
-		logger.Info("Successfully sent message to subscriber service", zap.String(responseStatus, response.Status))
+		logger.Info("Successfully sent message to subscriber service", zap.Int("statusCode", statusCode))
```

Comment:
Too chatty and not really valuable except for development right?

Reply:
Well it mirrors what we had before.. I think its ok for an info level.. but lets discuss if you disagree.

Reply:
After discussing, changed to debug.